### PR TITLE
[OneDrive] Add e2e feature

### PR DESCRIPTION
**Pull request by praveen-elastic**
## Closes https://github.com/elastic/connectors-py/issues/1415

This PR adds feature for end to end tests in OneDrive connector.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: tests/sources/test_onedrive.py [hunk 1]
```diff
@@ -60,54 +60,54 @@ def test_get_configuration():
 async def test_validate_configuration_with_invalid_dependency_fields_raises_error(
     extras,
 ):
-    source = create_source(OneDriveDataSource, **extras)
-
-    with pytest.raises(ConfigurableFieldValueError):
-        await source.validate_config()
+    async with create_source(OneDriveDataSource, **extras) as source:
```
Comment (wangch079) [inline tests/sources/test_onedrive.py:63]: Changes from this file seems not related to e2e, and I see them also changed in PR #1414

**Review by wangch079 — APPROVED** [review]
LGTM, just on nit.

File: .buildkite/nightly_steps.yml [hunk 2]
```diff
@@ -96,3 +96,9 @@ steps:
       - ".buildkite/run_nigthly.sh dropbox"
     artifact_paths:
       - "perf8-report-*/**/*"
+  - label: "🔨 OneDrive"
+    command:
+      - ".buildkite/run_nigthly.sh onedrive"
+    artifact_paths:
+      - "perf8-report-*/**/*"
```
Comment (artem-shelkovnikov) [inline .buildkite/nightly_steps.yml:103]: Need to also add it to `.buildkite/pipeline.yml` following the pattern that connectors follow there

File: tests/sources/fixtures/onedrive/fixture.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+app = Flask(__name__)
+
+
+DATA_SIZE = os.environ.get("DATA_SIZE", "small").lower()
+
+match DATA_SIZE:
+    case "extra_small":
+        TOTAL_USERS = 50
+        FILE_COUNT_PER_USER = 100
```
Comment (artem-shelkovnikov) [inline tests/sources/fixtures/onedrive/fixture.py:27]: Let's get rid of this

Comment (artem-shelkovnikov) [reply tests/sources/fixtures/onedrive/fixture.py:27]: (for context, I'm working on improving ftests and never had a need for extra_small)

**Review by artem-shelkovnikov — CHANGES_REQUESTED** [review]

File: .buildkite/nightly_steps.yml (hunk 2 above)
Comment (praveen-elastic) [reply .buildkite/nightly_steps.yml:103]: Thanks, Done.

**Review by artem-shelkovnikov — APPROVED** [review]
